monitors/test/SyntheticPegMonitor.js

```diff
@@ -15,92 +15,204 @@ contract("SyntheticPegMonitor", function(accounts) {
+        toBN(toWei("0.3")).toString()
+      );
+
+      // Test when volatility window captures only one historical price. The last update time is 100,
+      // so this should read the volatility from timestamps [100]. The min/max should be 10/10,
+      // and the volatility should be 0%.
+      medianizerPriceFeedMock.setLastUpdateTime(100);
+      assert.equal(
+        syntheticPegMonitor._calculateHistoricalVolatility(medianizerPriceFeedMock, 100, volatilityWindow).toString(),
+        "0"
+      );
+
+      // Test when volatility window captures only one historical price. The last update time is 200,
+      // so this should read the volatility from no timestamps. This should return null.
+      medianizerPriceFeedMock.setLastUpdateTime(200);
+      assert.equal(
+        syntheticPegMonitor._calculateHistoricalVolatility(medianizerPriceFeedMock, 200, volatilityWindow),
+        null
+      );
+
+      // Test when volatility window is smaller than the amount of historical prices. The last update time is 106,
+      // so this should read the volatility from timestamps [106, 105, 104, 103, 102]. The min/max should be 12/16,
+      // and the volatility should be (4 / 12 = 0.3333) or 33%.
+      medianizerPriceFeedMock.setLastUpdateTime(106);
+      assert.equal(
+        syntheticPegMonitor._calculateHistoricalVolatility(medianizerPriceFeedMock, 106, volatilityWindow).toString(),
+        toBN(toWei("0.333333333333333333")).toString() // 18 3's is max that can be represented with Wei.
+      );
+    });
+
+    it("Correctly emits messages", async function() {
+      // Inject prices into pricefeed.
+      const historicalPrices = [
+        { timestamp: 100, price: toBN(toWei("10")) },
+        { timestamp: 101, price: toBN(toWei("11")) },
+        { timestamp: 102, price: toBN(toWei("12")) },
+        { timestamp: 103, price: toBN(toWei("13")) },
+        { timestamp: 104, price: toBN(toWei("14")) },
+        { timestamp: 105, price: toBN(toWei("15")) },
+        { timestamp: 106, price: toBN(toWei("16")) },
+        { timestamp: 107, price: toBN(toWei("17")) }
+      ];
+      medianizerPriceFeedMock.setHistoricalPrices(historicalPrices);
+
+      // Test when volatility is under threshold. Monitor should not emit any events.
+      // Min/Max from time 103 should be 10/13, so volatility should be 3/10 = 30%, which is
+      // not greater than the 30% threshold.
+      medianizerPriceFeedMock.setLastUpdateTime(103);
+      await syntheticPegMonitor.checkPriceVolatility();
+      assert.equal(spy.callCount, 0); // There should be no messages sent at this point.
+
+      // Test when volatility is over threshold. Monitor should emit an error message.
+      // Min/Max from time 104 is 10/14, so volatility is 4/10 = 40%.
+      medianizerPriceFeedMock.setLastUpdateTime(104);
+      await syntheticPegMonitor.checkPriceVolatility();
+      assert.equal(spy.callCount, 1); // There should be no messages sent at this point.
```

Comment:
This comment is wrong.
```suggestion
      assert.equal(spy.callCount, 1); // There should be one messages sent at this point.
```